clmm/modbackend/cluster_toolkit.py

```diff
@@ -1,24 +1,24 @@
 # Functions to model halo profiles
 
 import cluster_toolkit as ct
+
 import numpy as np
+import warnings
 
 from astropy import units
 from astropy.cosmology import LambdaCDM, FlatLambdaCDM
 
 from .. constants import Constants as const
-from . import generic
-from . generic import get_reduced_shear_from_convergence
+from . import func_layer
+from . func_layer import *
 
 from .. clmm_cosmo import CLMMCosmology
+from .. clmm_modeling import CLMModeling
 
-import warnings
+from .ccl import CCLCosmology
+from .numcosmo import NumCosmoCosmology
```

Comment:
Not sure I understand while these need to be imported `cluster_toolkit` module.

Reply:
@m-aguena added support for converting between different cosmologies. So, these modules are imported because it checks if the cosmology is "convertible" looking at the class it comes from (so it needs to import CCL and NumCosmo). 

I think that this feature can be a bit dangerous. The conversion does not guarantee that the output cosmology will be exactly the same (we would have to mess with the other aspects of the background cosmologies, such as neutrino masses, Yb, reionization redshift, etc). Thus, the way it is now if you pass a NumCosmo cosmology to a CT object it will just silently convert by using the same values of the Omegas to create an AstroPy cosmology object but will ignore other cosmological parameters. If we are going to keep this feature I suggest adding a big warning every time such conversion happens.

Reply:
these two were imported so `CTModeling.set_cosmo` could check on the types of cosmology to be accepted. But you are right, it does impose to have `ccl` installed. I will implement a better way to check it without having those imports.